### PR TITLE
Example of using an European National ID card for client auth

### PR DESCRIPTION
**Pull request by yschimke**
This example works with the Estonian National ID Card. 

Output from s_server test server

```
Client certificate
Certificate:
    Data:
        Version: 3 (0x2)
        Serial Number:
            ...
        Signature Algorithm: sha256WithRSAEncryption
        Issuer: C=EE, O=AS Sertifitseerimiskeskus, CN=ESTEID-SK 2011/emailAddress=pki@sk.ee
        Validity
            Not Before: ...
            Not After : ...
        Subject: C=EE, O=ESTEID, OU=authentication, CN=...
```

### CODE REVIEW

File: samples/guide/src/main/java/okhttp3/recipes/CustomTrust.java [hunk 1]
```diff
@@ -171,7 +171,7 @@ private InputStream trustedCertificatesInputStream() {
    * not use custom trusted certificates in production without the blessing of your server's TLS
    * administrator.
    */
-  private X509TrustManager trustManagerForCertificates(InputStream in)
+  static X509TrustManager trustManagerForCertificates(InputStream in)
```
Comment (swankjesse) [inline samples/guide/src/main/java/okhttp3/recipes/CustomTrust.java]: (these samples are intended to be completely standalone)

File: samples/guide/src/main/java/okhttp3/recipes/OpenScClientAuthentication.java [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+import javax.net.ssl.KeyManager;
+import javax.net.ssl.KeyManagerFactory;
+import javax.net.ssl.SSLContext;
+import javax.net.ssl.SSLSocketFactory;
+import javax.net.ssl.TrustManager;
+import javax.net.ssl.X509TrustManager;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.Response;
+import sun.security.pkcs11.SunPKCS11;
```
Comment (swankjesse) [inline samples/guide/src/main/java/okhttp3/recipes/OpenScClientAuthentication.java]: this is not cross platform right?

**Comment by swankjesse** [comment]
I don’t think this is useful as an example, but it’s certainly useful. Maybe create a standalone GitHub project?

**Comment by yschimke** [comment]
Yeah, I've got it in my own project already.  I thought it was likely that it falls in between OpenSC and various clients, with no good home as a reference.

**Pull request closed without merging**